### ISSUE TITLE
How exactly does training on TPU works?

### ISSUE THREAD

**Issue by EnyMan** [tensorflow/tpu#325]
Hello, I have a few questions about TPUs and how the training works.

I am planning to use inceptionV3 to create embeddings of some pictures, with triplet loss. But I am having a hard time making the `input_fn` make proper triplets so, for now, it just returns the images "randomly". But I discovered an issue. Each core of the TPU gets part of the batch.

And my first question is what is exactly calculated on the TPU? Only the CNN logits and the rest is done on the host machine? Or is everything even the loss done on the TPUs core?

Also about evaluation, the `eval_metrics` is calculated at the end of the evaluation or after each batch? (just want to confirm what is written in `tf.contrib.tpu.TPUEstimatorSpec` documentation)

**Comment by saberkun** [comment]
Each batch will be distributed to each TPU cores. In a TPU donut, there are 4 chips with 8 cores.

The loss and gradients are calculated on Tensor cores and all reduce ops happens. Each core will have the same variable values.

Eval metrics are calculated in a different eval pass. After each batch, if you want to get something, you can implement a host call.

**Comment by EnyMan** [comment]
4 chips with 8 cores? So the google cloud v2-8 node is 4 chips each with 8 cores or one chip with 8 cores?

About the loss lets say i have batchsize of 1024 each core gets 1024/8 samples and calculates the loss on this portion? Then the loss is reduced to make the final loss? Am i understanding this correctly?

**Comment by EnyMan** [comment]
To answer its 8 cores in the v2-8, the Eval is done on the whole batch after all cores are done. And from what I gathered the loss is calculated per cores batch.
